src/py/flwr/server/state/sqlite_state.py

```diff
@@ -457,13 +471,35 @@ def unregister_node(self, node_id: int) -> None:
         query = "DELETE FROM node WHERE node_id = :node_id;"
         self.query(query, {"node_id": node_id})
 
-    def get_nodes(self) -> Set[int]:
+    def get_nodes(self, workload_id: str) -> Set[int]:
         """Retrieve all currently stored node IDs as a set."""
+        # Validate workload ID
+        query = "SELECT COUNT(*) FROM workload WHERE workload_id = ?;"
+        if self.query(query, (workload_id,))[0]["COUNT(*)"] == 0:
+            log(ERROR, "`workload_id` is invalid")
+            return set()
```

Comment:
What's the purpose of passing the `workload_id` into the function?

Additionally, if the goal of the function is to return all existing `node_ids` it should not expose a method to probe for valid workload_ids. That would be a security risk.

Was the goal of this change to get all `node_ids` which are related to a certain `workload_id`?

Reply:
Yes. Including `workload_id` in the `GetNodesRequest` and validating it is suggested by @danieljanes to prevent access from invalid workload and also to return different lists of nodes based on `workload_id` in the future.

Btw, I am a bit confused about the security risk you just mentioned. Would you mind elaborating on why workload validation can possibly be a security risk when the aim of the function `get_nodes()` is to return all existing nodes?

Reply:
I am unsure if I misread yesterday, but I thought we should not allow probing through the call to `get_nodes` if a `workload_id` exists. The return value in both of the following cases should be an empty `Set`:

Case 1: `workload_id` does not exist
Case 2: `workload_id` has no nodes